### PR TITLE
Config panel for ZHA

### PR DESCRIPTION
**Pull request by dmulcahey**
This PR adds the base for a config panel for the ZHA component. Supported functionality:

- permit joining - (was already a service, just added to panel)

### CODE REVIEW

**Comment by dmulcahey** [comment]
Not sure what the travis issue is... If someone is willing to point me in the right direction I'll correct the issue(s)...

File: src/panels/config/zha/ha-config-zha.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+import { PolymerElement } from "@polymer/polymer/polymer-element";
+
+import "../../../layouts/ha-app-layout";
+import "../../../resources/ha-style";
+
+import "./zha-network";
+import "./zha-node";
+
+/*
+ * @appliesMixin LocalizeMixin
```
Comment (balloob) [inline src/panels/config/zha/ha-config-zha.js]: Remove this comment since you're not applying the mixin.

File: src/panels/config/zha/zha-node.js [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+      .then((clusters) => {
+        this.setProperties({ clusters: clusters });
+      });
+  }
+
+  selectedClusterChanged(selectedCluster) {
+    if (selectedCluster === -1) return;
+    this.selectedClusterAttribute = -1;
+    this.selectedClusterCommand = -1;
+    this.hass
```
Comment (balloob) [inline src/panels/config/zha/zha-node.js]: Use async functions and then call `await this.hass.callWS(…)`

File: src/panels/config/zha/zha-node.js [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+      .callWS({
+        type: "zha/entities/clusters/attributes",
```
Comment (balloob) [inline src/panels/config/zha/zha-node.js]: Please move all these commands to `src/data/zha.ts`. See `src/data/shopping-list.ts` as an example.

File: src/resources/ha-style.js [hunk 4]
```diff
@@ -188,12 +188,19 @@ documentContainer.innerHTML = `<custom-style>
...
+        color: var(--primary-color);
+        font-weight: 500;
+      }
 
       .card-actions paper-button.warning:not([disabled]),
       .card-actions ha-call-api-button.warning:not([disabled]),
       .card-actions ha-call-service-button.warning:not([disabled]) {
         color: var(--google-red-500);
       }
+      .card-actions ha-call-ws-api-button.warning:not([disabled]) {
```
Comment (balloob) [inline src/resources/ha-style.js]: Please add this declaration to the previous one.

File: src/resources/ha-style.js [hunk 5]
```diff
@@ -188,12 +188,19 @@ documentContainer.innerHTML = `<custom-style>
         color: var(--primary-color);
         font-weight: 500;
       }
+      .card-actions ha-call-ws-api-button:not([disabled]) {
```
Comment (balloob) [inline src/resources/ha-style.js]: Please add this declaration to the previous one.

**Comment by dmulcahey** [comment]
Point me towards a good example using Lit elements and I’ll redo the entire set of changes that way (but in smaller PR’s)

File: src/components/buttons/ha-call-ws-api-button.js [hunk 6]
```diff
@@ -0,0 +1,62 @@
+import { LitElement, html } from "@polymer/lit-element";
+
+import "./ha-progress-button";
+import { fireEvent } from "../../common/dom/fire_event";
+
+class HaCallApiButton extends LitElement {
```
[the remark this answers is not in the document]
Comment (dmulcahey) [reply src/components/buttons/ha-call-ws-api-button.js]: Is this something you want me to build?

File: src/resources/ha-style.js (hunk 4 above)
Comment (dmulcahey) [reply src/resources/ha-style.js]: Not sure I follow this comment... also, does this still apply with the other button comment?

Comment (iantrich) [reply src/resources/ha-style.js]: No need for a second style that does the same thing
```suggestion
      .card-actions paper-button.warning:not([disabled]),
      .card-actions ha-call-api-button.warning:not([disabled]),
      .card-actions ha-call-service-button.warning:not([disabled]),
      .card-actions ha-call-ws-api-button.warning:not([disabled]) {
        color: var(--google-red-500);
      }
```

**Comment by iantrich** [comment]
> Point me towards a good example using Lit elements and I’ll redo the entire set of changes that way (but in smaller PR’s)

Lots of examples of TypeScript and LitElement in [lovelace](https://github.com/home-assistant/home-assistant-polymer/tree/dev/src/panels/lovelace)

File: src/resources/ha-style.js (hunk 4 above)
Comment (dmulcahey) [reply src/resources/ha-style.js]: Awesome, that makes sense now. Thx

File: src/panels/config/zha/ha-config-zha.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import "@polymer/app-layout/app-header/app-header";
```
Comment (balloob) [inline src/panels/config/zha/ha-config-zha.js]: Should be a TS file.

**Review by balloob — APPROVED** [review]

**Comment by balloob** [comment]
ok to merge when last comment addressed.

File: src/panels/config/zha/ha-config-zha.ts [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+          .children[0] as HTMLTemplateElement).content,
+        true
+      );
+    }
+    return html`
+      ${this._ironFlex} ${this._haStyle}
+    `;
+  }
+
+  private _onBackTapped() {
```
Comment (iantrich) [inline src/panels/config/zha/ha-config-zha.ts]:
```suggestion
  private _onBackTapped(): void {
```

**Pull request merged**